core/shared/src/main/scala/fs2/interop/flow/StreamSubscription.scala

```diff
@@ -192,9 +192,8 @@ private[flow] object StreamSubscription {
 
   def subscribe[F[_], A](stream: Stream[F, A], subscriber: Subscriber[A])(implicit
       F: Async[F]
-  ): F[Unit] =
-    apply(stream, subscriber).flatMap { subscription =>
-      F.delay(subscriber.onSubscribe(subscription)) >>
-        subscription.run
+  ): Stream[F, Nothing] =
+    Stream.eval(apply(stream, subscriber)).flatMap { subscription =>
+      Stream.eval(F.delay(subscriber.onSubscribe(subscription))) >> subscription.run
```

Comment:
This should be `exec`, shouldn't it?

Reply:
If it's exec, then `subscription.run` will never evaluate.

The two options are:
```scala
Stream.eval(...) >> subscription.run
Stream.exec(...) ++ subscription.run
```
The former felt more appropriate to me, it establishes the monadic dependence. Not sure what the subtle differences are.